pkg/cli/testdata/MockClient.go

```diff
@@ -407,6 +407,41 @@ func (m *MockClient) ImageGet(ctx context.Context, name string) (*apiv1.Image, e
 		return &apiv1.Image{}, nil
 	case "found.image-two-tags":
 		return &apiv1.Image{}, nil
+	case "testtag":
```

Comment:
How would you feel about adding a similar functionality here as in the `ImageList` mock? That way, you could provide an image to the `MockClientFactory` for a test and this function would return it. My fear is that this `switch` would get unruly for all the tests we would write.

Reply:
I was planning to open a separate PR for a mini refactor on the mockclient as there were some other cli changes needed as well. Agreed that it will get hard to manage.

Reply:
Sounds good.